Firestore/Source/API/FIRDocumentSnapshot.mm

```diff
@@ -173,16 +178,18 @@ - (id)convertedValue:(FSTFieldValue *)value options:(FSTFieldValueOptions *)opti
     return [self convertedArray:(FSTArrayValue *)value options:options];
   } else if ([value isKindOfClass:[FSTReferenceValue class]]) {
     FSTReferenceValue *ref = (FSTReferenceValue *)value;
-    FSTDatabaseID *refDatabase = ref.databaseID;
-    FSTDatabaseID *database = self.firestore.databaseID;
-    if (![refDatabase isEqualToDatabaseId:database]) {
+    const DatabaseId *refDatabase = ref.databaseID;
+    const DatabaseId *database = self.firestore.databaseID;
+    if (refDatabase != database) {
```

Comment:
This is now comparing pointer equality. Don't want this to be `*refDatabase != *database`?

Reply:
Good catch! Thanks.